### PR TITLE
implement plink2/score module

### LINKED ISSUE #1229: new module: plink2/score
I think it would be good to have a module for [plink2/score](https://bioconda.github.io/recipes/plink2/README.html)

This module would be useful for anybody looking to do [polygenic risk scoring](https://www.cog-genomics.org/plink/2.0/score). Also, plink2 scoring is significantly faster than plink1 scoring because it does matrix multiplication. 

- [X] This module does not exist yet with the [`nf-core modules list`](https://github.com/nf-core/tools#list-modules) command
- [X] There is no [open pull request](https://github.com/nf-core/modules/pulls) for this module
- [X] There is no [open issue](https://github.com/nf-core/modules/issues) for this module
- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by nebfield**
## PR checklist

Closes #1229 

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, include test data in your PR.
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [X] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [X] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [X] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/plink2/score/main.nf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::plink2=2.00a2.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/plink2:2.00a2.3--h712d239_1' :
+        'quay.io/biocontainers/plink2:2.00a2.3--h712d239_1' }"
+
+    input:
+    tuple val(meta), path(pgen), path(psam), path(pvar)
+    path scorefile
```
Comment (sateeshperi) [inline modules/plink2/score/main.nf:12]: consider adding () for consistency `path (scorefile)`

**Review by sateeshperi — COMMENTED** [review]
rest lgtm

File: modules/plink2/score/main.nf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    output:
+    tuple val(meta), path("*.sscore"), emit: score
+    path "versions.yml"              , emit: versions
+
+    script:
```
Comment (sateeshperi) [inline modules/plink2/score/main.nf:18]:
```
    when:
    task.ext.when == null || task.ext.when
````
needs this

File: modules/plink2/score/main.nf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    def mem_mb = task.memory.toMega() // plink is greedy
+    """
+    plink2 \\
+        --threads $task.cpus \\
+        --memory $mem_mb \\
+        --pfile ${pgen.baseName} \\
+        --score ${scorefile} \\
+        $args \\
+        --out ${prefix}
+
```
[the remark this answers is not in the document]
Comment (nebfield) [reply modules/plink2/score/main.nf:30]: Support for compressing text files is buggy in this version of plink. The bug has been fixed in the development version, but won't be available as a container for a while :(

**Review by sateeshperi — APPROVED** [review]
lgtm

**Pull request merged**